engine/impl/Playlist.php

```diff
@@ -440,40 +454,56 @@ public function isDateTimeWithinShow($timeStamp, $listRow) {
 
     // insert playlist track. return following: 0 - fail, 1 - success no 
     // timestamp, 2 - sucess with timestamp.
-    public function insertTrack($playlistId, $tag, $artist, $track, $album, $label, $wantTimestamp, &$id = null) {
+    public function insertTrack($playlistId, $tag, $artist, $track, $album, $label,  $insertTime, &$id, &$status) {
         $row = $this->getPlaylist($playlistId, 1);
 
-        // log time iff 'now' is within playlist start/end time.
-        $doTimestamp = $wantTimestamp && $this->isNowWithinShow($row);
-        $timeName    = $doTimestamp ? "created, " : "";
-        $timeValue   = $doTimestamp ? "NOW(), "   : "";
+        $isWithinShow = $this->isWithinShow($insertTime, $row);
+        if (!$isWithinShow) {
+            $status = "Spin time is outside of show start/end times. ";
+            error_log($status);
+            return 0;
+        }
+
+        // abort if the insert time is not greater than the last entry time.
+        // this is needed because the sequence reorder does not work in this
+        // case. if this check is removed then this issue must be fixed.
+        if ($insertTime != null) {
```

Comment:
Actually, at the very end of this function, there is a test made to see if the inserted track's timestamp is last or not, and if not, it does a reorder.  The sequence of the newly inserted track propagates back to the caller, who does the right thing in the javascript code to put the new track in the right spot in the list.  If this *isn't* working, let me know, I will look into it.

Reply:
Yes, I encountered inserts where the seq value in the DB was still incorrect even after reorderForTime() was invoked.